app/cart/controller.js

```diff
@@ -2,12 +2,16 @@ import { alias } from '@ember/object/computed';
 import Controller from '@ember/controller';
 
 export default Controller.extend({
-  // cartItems: Ember.computed.map('model', function(item) {
-  //   return {
-  //     id:
-  //     name: item.get('product.name'),
-  //     quantity: item.get('quantity')
-  //   };
-  // })
-  cartItems: alias('model')
+  totalPrice: Ember.computed('cartItemCount', function() {
+    const cartItemCount = this.get('cartItemCount') * 25
+    return `$${cartItemCount}.00`
+  }),
+  cartItemCount: Ember.computed('model.@each.quantity', function() {
+    if (this.get('model.length')) {
+      const quantities = this.get('model').mapBy('quantity');
+      return quantities.reduce((a, b) => parseInt(a, 10) + parseInt(b, 10));
+    }
+
+    return 0;
+  }),
```

Comment:
can avoid some code:

```js
import { mapBy, sum } from '@ember/object/computed';
```
```js
cartItemQuantities: mapBy('model', 'quantity'),
cartItemCount: sum('cartItemQuantities')
```

Reply:
👍  will try this

Reply:
@jmonster this is a clean way of doing it, but it doesn't seem to parse the values, which in turn, concats the values if they're strings instead of adding them up. i've been using a "vanilla" implementation of select:
```
<select onchange={{action (mut cartItem.quantity) value="target.value"}} class="" name="">
                <option selected={{eq cartItem.quantity 1}} value="1">1</option>
                <option selected={{eq cartItem.quantity 2}} value="2">2</option>
                <option selected={{eq cartItem.quantity 3}} value="3">3</option>
                <option selected={{eq cartItem.quantity 4}} value="4">4</option>
                <option selected={{eq cartItem.quantity 5}} value="5">5</option>
              </select>
```

ex. select 2 as the quantity
this updates the value of the record quantity to `2`, but its value is a string. maybe i need to change this to use `x-select` instead?